bndtools.core/src/org/bndtools/core/templating/repobased/ReposTemplateLoader.java

```diff
@@ -121,6 +121,9 @@ void dectivate() {
         // Map List<Repository> to Promise<List<Template>>
         Promise<List<Template>> promise = repos.stream().map(repo -> promiseFactory.submit(() -> {
             Map<Requirement,Collection<Capability>> providerMap = repo.findProviders(Collections.singleton(requirement));
+            if (!providerMap.containsKey(requirement)) {
```

Comment:
This is not possible. Per the specification of the Repository.findProviders method:

```If there are no matching capabilities for a specified requirement, then the value in the map for the specified requirement must be an empty collection.```

So `requirement` MUST be a key in the returned map.

What is the NPE stacktrace?

Reply:
So I was using the current bndtools/workspace template (3.5) and the InfoRepositoryWrapper around the JPM repo was causing the problem. So this may be a bnd issue. I am looking now.